TypeCobol/Compiler/Parser/CodeElementsParserStep.cs

```diff
@@ -332,6 +333,49 @@ internal static IList<DocumentChange<ICodeElementsLine>> ParseProcessedTokensLin
                                     MessageCode.TypeCobolParserLimitation);
                             }
 
+                            // check if End statement is aligned with the matching opening statement
+                            if (checkEndAlignment && codeElement.Type.ToString().EndsWith("End"))
+                            {
+                                string endStatement = codeElement.Type.ToString();
+                                if (!endStatement.Equals("ProgramEnd", StringComparison.OrdinalIgnoreCase))
+                                {
+                                    string statement = endStatement.Substring(0, endStatement.Length - 3);
+                                    // search code elements from more recent to the less recent codeElementsLinesChanges
+                                    var list = codeElementsLinesChanges?
+                                        .Where(cel => cel.NewLine.CodeElements != null)?
```

Comment:
Null check is redundant. LINQ will produce a non-null empty enumerable if no items match the criteria.

Reply:
deleted
https://github.com/TypeCobolTeam/TypeCobol/tree/443efbd3422c032e41827e7c89979c9ddfa2bb44